middleware/http/transport.go

```diff
@@ -154,6 +154,15 @@ func (t *transport) RoundTrip(req *http.Request) (res *http.Response, err error)
 		req.Context(), req.URL.Scheme+"/"+req.Method, zipkin.Kind(model.Client), zipkin.RemoteEndpoint(t.remoteEndpoint),
 	)
 
+	// inject whitelisted headers from spancontext into the outgoing HTTP request headers
+	if sp.Context().Baggage != nil {
+		sp.Context().Baggage.IterateHeaders(func(key string, vals []string) {
```

Comment:
could IterateHeaders handle the nil pointer case and remove indent here and elsewhere by doing an early exit? or would you consider it more efficient to do this at each call site (not sure the stack cost blah blah).

Feels like handling nil ref is something you can do later, though switching from non-nil only to maybe nil not.

Reply:
Yes I could add in current implementation but didn't as the Interface does not enforce this. Since this is pluggable one could write an implementation that doesn't check for nil and it will panic if I remove the test here. This way it is not as fragile.